### ISSUE TITLE
Exception when using AddMultitenantGrainStorage

### ISSUE THREAD

**Issue by ErikAllan** [VincentH-Net/Orleans.Multitenant#1]
Hi!

I don't know if this is a bug, but I get an exception when I try to use the AddMultitenantGrainStorage method. This is my code:

```
siloBuilder.AddMultitenantGrainStorage<AzureTableGrainStorage, AzureTableStorageOptions, AzureTableGrainStorageOptionsValidator>(
            "order", (silo, name) =>
            {
                return silo.AddAzureTableGrainStorage(name, options => options.ConfigureTableServiceClient("UseDevelopmentStorage=true"));
            },
            configureTenantOptions: (options, tenantId) =>
            {
                options.ConfigureTableServiceClient("UseDevelopmentStorage=true");
                options.TableName = $"Order{tenantId}";
            });
```

**Exception (raised at startup):** System.InvalidOperationException: 'Unable to resolve service for type 'Orleans.Multitenant.Internal.ITenantGrainStorageFactory' while attempting to activate 'Orleans.Multitenant.Internal.MultitenantStorage'.'

Call stack:

 	[Exception] Microsoft.Extensions.DependencyInjection.Abstractions.dll!Microsoft.Extensions.DependencyInjection.ActivatorUtilities.ConstructorMatcher.CreateInstance(System.IServiceProvider provider)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.Abstractions.dll!Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(System.IServiceProvider provider, System.Type instanceType, object[] parameters)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.Abstractions.dll!Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance<T>(System.IServiceProvider provider, object[] parameters)	Unknown
 	[Exception] Orleans.Multitenant.dll!Orleans.Multitenant.Internal.MultitenantStorageFactory.Create(System.IServiceProvider services, string name)	Unknown
>	[Exception] Orleans.Core.dll!Orleans.Runtime.KeyedSingletonService<TKey, TService>..ctor.AnonymousMethod__0() Line 83	C#
 	[Exception] System.Private.CoreLib.dll!System.Lazy<T>.ViaFactory(System.Threading.LazyThreadSafetyMode mode)	Unknown
 	[Exception] System.Private.CoreLib.dll!System.Lazy<T>.ExecutionAndPublication(System.LazyHelper executionAndPublication, bool useDefaultConstructor)	Unknown
 	[Exception] System.Private.CoreLib.dll!System.Lazy<T>.CreateValue()	Unknown
 	[Exception] Orleans.Core.dll!Orleans.Runtime.KeyedSingletonService<TKey, TService>.GetService(System.IServiceProvider services) Line 87	C#
 	[Exception] Orleans.Core.dll!Orleans.Runtime.KeyedServiceCollection<TKey, TService>.GetService(System.IServiceProvider services, TKey key) Line 125	C#
 	[Exception] Orleans.Core.Abstractions.dll!Orleans.Runtime.KeyedServiceCollectionExtensions.GetServiceByKey<TKey, TService>(System.IServiceProvider services, TKey key) Line 78	C#
 	[Exception] Orleans.Core.Abstractions.dll!Orleans.Runtime.KeyedServiceCollectionExtensions.GetRequiredServiceByKey<TKey, TService>(System.IServiceProvider services, TKey key) Line 100	C#
 	[Exception] Orleans.Core.Abstractions.dll!Orleans.Runtime.KeyedServiceCollectionExtensions.GetRequiredServiceByName<TService>(System.IServiceProvider services, string name) Line 138	C#
 	[Exception] Orleans.Multitenant.dll!Orleans.Multitenant.Internal.SiloBuilderExtensions.AddMultitenantGrainStorage.AnonymousMethod__0_3(System.IServiceProvider s, string n)	Unknown
 	[Exception] Orleans.Core.dll!Orleans.Runtime.KeyedSingletonService<TKey, TService>..ctor.AnonymousMethod__0() Line 83	C#
 	[Exception] System.Private.CoreLib.dll!System.Lazy<T>.ViaFactory(System.Threading.LazyThreadSafetyMode mode)	Unknown
 	[Exception] System.Private.CoreLib.dll!System.Lazy<T>.ExecutionAndPublication(System.LazyHelper executionAndPublication, bool useDefaultConstructor)	Unknown
 	[Exception] System.Private.CoreLib.dll!System.Lazy<T>.CreateValue()	Unknown
 	[Exception] Orleans.Core.dll!Orleans.Runtime.KeyedSingletonService<TKey, TService>.GetService(System.IServiceProvider services) Line 87	C#
 	[Exception] Orleans.Core.dll!Orleans.Runtime.KeyedServiceCollection<TKey, TService>.GetService(System.IServiceProvider services, TKey key) Line 125	C#
 	[Exception] Orleans.Core.Abstractions.dll!Orleans.Runtime.KeyedServiceCollectionExtensions.GetServiceByKey<TKey, TService>(System.IServiceProvider services, TKey key) Line 78	C#
 	[Exception] Orleans.Core.Abstractions.dll!Orleans.Runtime.KeyedServiceCollectionExtensions.GetRequiredServiceByKey<TKey, TService>(System.IServiceProvider services, TKey key) Line 100	C#
 	[Exception] Orleans.Core.Abstractions.dll!Orleans.Runtime.KeyedServiceCollectionExtensions.GetRequiredServiceByName<TService>(System.IServiceProvider services, string name) Line 138	C#
 	[Exception] Orleans.Persistence.AzureStorage.dll!Orleans.Hosting.AzureTableSiloBuilderExtensions.AddAzureTableGrainStorage.AnonymousMethod__4_2(System.IServiceProvider s, string n) Line 60	C#
 	[Exception] Orleans.Core.dll!Orleans.Runtime.KeyedSingletonService<TKey, TService>..ctor.AnonymousMethod__0() Line 83	C#
 	[Exception] System.Private.CoreLib.dll!System.Lazy<T>.ViaFactory(System.Threading.LazyThreadSafetyMode mode)	Unknown
 	[Exception] System.Private.CoreLib.dll!System.Lazy<T>.ExecutionAndPublication(System.LazyHelper executionAndPublication, bool useDefaultConstructor)	Unknown
 	[Exception] System.Private.CoreLib.dll!System.Lazy<T>.CreateValue()	Unknown
 	[Exception] Orleans.Core.dll!Orleans.Runtime.KeyedSingletonService<TKey, TService>.GetService(System.IServiceProvider services) Line 87	C#
 	[Exception] Orleans.Runtime.dll!Orleans.Runtime.Silo..ctor.AnonymousMethod__43_0(Orleans.Runtime.IKeyedService<string, Orleans.ILifecycleParticipant<Orleans.Runtime.ISiloLifecycle>> s) Line 181	C#
 	[Exception] System.Linq.dll!System.Linq.Enumerable.SelectArrayIterator<TSource, TResult>.MoveNext()	Unknown
 	[Exception] Orleans.Runtime.dll!Orleans.Runtime.Silo.Silo(Orleans.Runtime.ILocalSiloDetails siloDetails, System.IServiceProvider services) Line 183	C#
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite constructorCallSite, Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext context)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor<TArgument, TResult>.VisitCallSiteMain(Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceCallSite callSite, TArgument argument)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceCallSite callSite, Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext context)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor<TArgument, TResult>.VisitCallSite(Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceCallSite callSite, TArgument argument)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitConstructor(Microsoft.Extensions.DependencyInjection.ServiceLookup.ConstructorCallSite constructorCallSite, Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext context)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor<TArgument, TResult>.VisitCallSiteMain(Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceCallSite callSite, TArgument argument)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceCallSite callSite, Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext context)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor<TArgument, TResult>.VisitCallSite(Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceCallSite callSite, TArgument argument)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitIEnumerable(Microsoft.Extensions.DependencyInjection.ServiceLookup.IEnumerableCallSite enumerableCallSite, Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext context)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor<TArgument, TResult>.VisitCallSiteMain(Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceCallSite callSite, TArgument argument)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.VisitRootCache(Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceCallSite callSite, Microsoft.Extensions.DependencyInjection.ServiceLookup.RuntimeResolverContext context)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteVisitor<TArgument, TResult>.VisitCallSite(Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceCallSite callSite, TArgument argument)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceLookup.CallSiteRuntimeResolver.Resolve(Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceCallSite callSite, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope scope)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceProvider.CreateServiceAccessor(System.Type serviceType)	Unknown
 	[Exception] System.Collections.Concurrent.dll!System.Collections.Concurrent.ConcurrentDictionary<TKey, TValue>.GetOrAdd(TKey key, System.Func<TKey, TValue> valueFactory)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.dll!Microsoft.Extensions.DependencyInjection.ServiceProvider.GetService(System.Type serviceType, Microsoft.Extensions.DependencyInjection.ServiceLookup.ServiceProviderEngineScope serviceProviderEngineScope)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.Abstractions.dll!Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService(System.IServiceProvider provider, System.Type serviceType)	Unknown
 	[Exception] Microsoft.Extensions.DependencyInjection.Abstractions.dll!Microsoft.Extensions.DependencyInjection.ServiceProviderServiceExtensions.GetRequiredService<T>(System.IServiceProvider provider)	Unknown
 	[Exception] Microsoft.Extensions.Hosting.dll!Microsoft.Extensions.Hosting.Internal.Host.StartAsync(System.Threading.CancellationToken cancellationToken)	Unknown
 	[Exception] Microsoft.Extensions.Hosting.Abstractions.dll!Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(Microsoft.Extensions.Hosting.IHost host, System.Threading.CancellationToken token)	Unknown
 	[Exception] Microsoft.Extensions.Hosting.Abstractions.dll!Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.RunAsync(Microsoft.Extensions.Hosting.IHost host, System.Threading.CancellationToken token)	Unknown
 	System.Private.CoreLib.dll!System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()	Unknown
 	System.Private.CoreLib.dll!System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(System.Threading.Tasks.Task task)	Unknown
 	System.Private.CoreLib.dll!System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(System.Threading.Tasks.Task task)	Unknown
 	[Waiting on Async Operation, double-click or press enter to view Async Call Stacks]	
 	Microsoft.Extensions.Hosting.Abstractions.dll!Microsoft.Extensions.Hosting.HostingAbstractionsHostExtensions.Run(Microsoft.Extensions.Hosting.IHost host)	Unknown
 	Silo.dll!Program.<Main>$(string[] args) Line 160	C#

Thank you for a great Orleans extension anyway :-)

/Erik

**Comment by VincentH-Net** [comment]
Fixed in [1.1.4](https://www.nuget.org/packages/Orleans.Multitenant/1.1.4)
Thanks for reporting @ErikAllan!

**Comment by ErikAllan** [comment]
Thank you for the patch! It seems to work fine now :-)
